### PR TITLE
Parse out citations from wikipedia and reuse them.

### LINKED ISSUE #17: Scrape citations from the wikipedia page
We're parsing wikipedia's list of house election candidates. Collect relevant citations from that page, and add them to the candidate information dictionary so that we can use them in creating pages.

### PR DESCRIPTION
**Pull request by whereistanya**
Fixes #17

### CODE REVIEW

**Comment by rdhyee** [comment]
What I needed to do to get `py.test` to be happy:
- assumed that code is Python 2.x
- install https://pypi.python.org/pypi/us in addition to other dependencies.

Haven't had a chance to review code yet, but this is my progress thus far.

**Comment by whereistanya** [comment]
@rdhyee Added 2.7 to the test_candidate.py shebang (at master), to match the main code. Updated the README.

**Comment by rdhyee** [comment]
What do you think is the current status of the code (after we merge this PR)?   From the readme:

> It can currently read yaml, the xml data dump from http://www.fec.gov/data/CandidateSummary.do and https://en.wikipedia.org/wiki/United_States_House_of_Representatives_elections,_2016

The title for this PR is "Parse out citations from wikipedia and reuse them".  Do you mean the code from this PR will parse out citations from https://en.wikipedia.org/wiki/United_States_House_of_Representatives_elections,_2016 -- or from some other page?  What does "reuse them" mean specifically?

**Comment by whereistanya** [comment]
After this CL, it reads a local copy of https://en.wikipedia.org/wiki/United_States_House_of_Representatives_elections,_2016. That has a list of candidates for election, each with a citation to show how we know they're a candidate. We grab that citation and store it, then use it in the page we create. See the "X is a Democratic candidate..." line at  http://cso.noidea.dog/w/index.php/User:Candidatebot/sandbox/Will_Boyd ? That [1] at the end is the citation we scraped from the House_of_Representatives page.

File: candidate.py [hunk 1]
```diff
@@ -317,28 +324,52 @@ def new_from_wikipedia_page(filename):
...
           if tags[i].name == 'a' and tags[i].text == name:
             page = tags[i]
         if page:  # The candidate already has a page. We're not needed here.
           continue
+
+        # Extract the citation. This is a bit involved.
+        # 1. We pull the reference number out from after the candidate name. It
+        # looks like:  [63]
+        # 2. We get the A record that matches that reference. It looks like:
+        #   <a href="#cite_note-68">[63]</a> The two numbers won't match, btw.
```
Comment (rdhyee) [inline candidate.py:339]: "The two numbers won't match" -> "The two numbers usually don't match" ?

**Review by rdhyee — COMMENTED** [review]
By "local copy", you mean https://github.com/DevProgress/candidatebot/blob/master/test_house.html? In my review, should I be looking at https://en.wikipedia.org/wiki/United_States_House_of_Representatives_elections,_2016 too and asking myself where your code might or might be able to handle the live Wikipedia page as opposed to the test data?

File: candidate.py (hunk 1 above)
Comment (whereistanya) [reply candidate.py:339]: Not worth a push for the comment fix. I'll fix it in the next commit.

**Comment by whereistanya** [comment]
Nope, just the local file. We can't control for major changes to the wikipedia page, so we'll just keep our local copy. I'm going to go ahead and commit. If you have any other comments, I'll fix them in a followup change.

**Pull request merged**